nav2_theta_star_planner/include/nav2_theta_star_planner/theta_star.hpp

```diff
@@ -0,0 +1,298 @@
+// Copyright 2020 Anshumaan Singh
```

Comment:
Why aren't more of these functions inline?

Reply:
There weren't a lot of functions that would benefit from it. Smaller functions like the ones that were calculating costs and were required to be called for a lot of iterations seemed the best in terms of being inlined, so I inlined only those functions

Reply:
Isn't `setStartAndGoal` `isUnsafeToPlan` `addIndex` `getIndex` potentially good candidates?

Reply:
`getIndex` could be inlined. But I am not really sure if `addIndex` should be inlined or not, as it sets the value of a global variable. As `setStartAndGoal` and `isUnsafeToPlan` are used only once during each execution of the code, I didnt inline it